### PR TITLE
fix autofill not working on page load

### PR DESCRIPTION
**Pull request by flo-mic**
I noticed that the autofill feature is not working on page load. Only on tab change event.

After some tests I figured out that the event fires before the login form is loaded, so it can not find the autofill controls of the webpage.

I added some delay for the autofill feature to solve the bug. I choosed 2,5 seconds for the delay. With this delay autofill  worked on most of my test pages. Only on pages where the connection was really slow autofill still failed.

### CODE REVIEW

File: src/js/Manager/AutofillManager.js [hunk 1]
```diff
@@ -27,13 +30,24 @@ export default new class AutofillManager {
...
         if(recommendations.length === 0 || await SettingsService.getValue('paste.autofill') === false) return;
         let password = recommendations[0];
+        var self = this;
 
-        let ids = TabManager.get('autofill.ids', []);
-        if(ids.indexOf(password.getId()) === -1) {
-            ids.push(password.getId());
-            TabManager.set('autofill.ids', ids);
-        }
+        setTimeout(function () {
```
Comment (marius-wieschollek) [inline src/js/Manager/AutofillManager.js:35]: You can actually use an arrow function here. This solves the problem with `this`.
```suggestion
        setTimeout(function () {
        setTimeout(() => {
```

**Comment by marius-wieschollek** [comment]
Thanks for investigating the issue.

The problem i see with the wait time is that for some websites it could not be long enough.
Also if we get the current result and insert it 2.5 seonds later, the page could have changed and we're sending the password off to a probably completely different url. This could be exploited if you find a page that has something like an "outgoing links page" that automatically redirects.

Maybe there is something like a "domcontentloaded" event we can attach to in the backend?

**Comment by flo-mic** [comment]
Hi  @marius-wieschollek, yes I agree with you. Maybe I could add some content script that informs the AutofillManager via Messageservice when the page is loaded completely. So we could ensure that the autofill will run in the right moment and on the right page. I will investigate on this again.

**Comment by flo-mic** [comment]
@marius-wieschollek the autofill feature is now working on every test page (100% success :-)). I added a event listener for the "domcontentloaded" event so that the AutofillManager gets informed that the page is ready.

There is still a timeout required because some client side javascript is still running when the event fires. But with 500 ms i was able to load every of my test pages. Also the one that failed before. And the password filling is incredible fast if I compare this with e.g. LastPass :)

Additionally I added some layer of verification. I'm storing the current url where the password recommendations belong to. The client also sends his current url. Only if both map the autofill will run. So we can avoid that the content script was initiated from another tab in the background or the user switches the page.

File: src/js/Manager/AutofillManager.js (hunk 1 above)
Comment (flo-mic) [reply src/js/Manager/AutofillManager.js]: Fixed

**Review by marius-wieschollek — APPROVED** [review]

**Pull request merged**